### DISCUSSION TITLE
[Errno None] b´libusb0-dll:err [_usb_reap_async] timeout error\n´

### DISCUSSION THREAD

**Discussion by MartinELG** [meerk40t/meerk40t#1938] [General]
Buen día, utilizo el programa MeerK40t con una maquina CO2 de tarjeta m2, un día anterior la use con normalidad y al día siguiente al quererla usar me arrojaba ese código **[Errno None] b´libusb0-dll:err [_usb_reap_async] timeout error\n´** y la maquina se detenía, y se empezaba a reconectar y desconectar en un ciclo continuo, y a parecer después perdía las coordenadas porque el equipo se ubicaba en una posición diferente a la marcada en el programa ¿A alguien le a pasado esto?

**Comment by Sophist-UK** [comment]
Someone else had a similar bug recently, reported in [#1931 libusb0-dll:err [control_msg] sending control message failed, win error: The system cannot find the file specified](https://github.com/meerk40t/meerk40t/issues/1931). Have you tried following the advice given there?

**Reply by MartinELG** [reply]
Si estuve verificando pero no me sirvió algo de lo mencionado ahí, creo como mencionan es problema del controlador M2, procedí a comprar uno nuevo, veré si funciona y les comparto el resultado! gracias!

**Comment by tatarize** [comment]
This is probably a hardware issue.

Can you download a copy of K40 Whisperer and see that that program either can control your laser. If MeerK40t doesn't work and K40 Whisperer does, then this is a certainly a problem with MeerK40t. If neither program works, this is a hardware problem or driver issue (most likely).

The error code there literally means that it tried to send the command but didn't hear back from the controller card, which could be just a one-off communication issue or could be a hardware issue.

**Reply by MartinELG** [reply]
Buen día compañero! probé con 3 programas diferentes (MeerK40t, K40 Whisperer, SagonLaser que es una Variación de K40Whisperer) y en todos me arrojaba el mismo problema, así que opte por encargar un controlador nuevo, para descartar lo mismo que menciona del hardware, les estaré comunicando si el problema se soluciono o persistió, muchas gracias

**Reply by tatarize** [reply]
It's likely the case that it's a hardware issue of some variety. I had this happen at one point when my y-axis was jammed, since it could never move my y-axis it would never trip the end-stop and since it never tripped the endstop during boot it never actually let you take control of the card. But, it could be a variety of other things. You could probably query the status on the card to see somewhat what's going on. But, in any event, it's not likely to be a software issue.

**Reply by Sophist-UK** [reply]
> You could probably query the status on the card to see somewhat what's going on.

If diagnostic capabilities are available on the board, it would be great if MK had some basic command line functionality which could query them.

**Reply by tatarize** [reply]
Well, you could see whether the board is connected and giving valid replies to various queries. If it's giving 239s that's generally considered booting and low-power at points. Whereas 238 means busy, which if it stays there means that the board is homing forever for some reason. And if you're getting pure 207s it's probably something seriously wrong with the communications. You can type `status` and it'll force query the board. Though it usually would be doing that a bit anyway. But `timer 0 0.2 .status` as a command is actually pretty useful for diagnostics at time. 

Oh, also there's some chance that maybe he has an M3-Nano and the hardware pause is somehow engaged. That would also trigger other and rather obvious signals that would flicker back and forth between those states.
